Test

sameerz09 opened the pull request:
Creating a pull request to configure the ODOO 16 API



---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

sameerz09 committed:
test adding API

sameerz09 committed:
testing

sameerz09 committed:
test

sameerz09 committed:
remove unncessary folder

sameerz09 committed:
update the test API

sameerz09 committed:
modified the API

sameerz09 committed:
Revert "modified the API"

This reverts commit af2e81c5bd130e96b2aa8eacc81911f8e919748c.

sameerz09 committed:
Updates made to the API

sameerz09 committed:
Delete addons/accounts_balances directory

sameerz09 committed:
Creating a pull request to configure the ODOO 16 API

sameerz09 committed:
Develop an API for calculating account movement balances and integrate it with third-party applications

sameerz09 committed:
Develop an account movement balances API enhancment and try to integrate it with third-party applications

sameerz09 committed:
Change the implementation from using the http.request method to utilizing an API method for data retrievals

sameerz09 committed:
Change the implementation from using the http.request method to utilizing an API method for data retrieval

sugampandey commented on addons/accounts_balances/demo/demo.xml:
```diff
@@ -0,0 +1,30 @@
+<odoo>
+    <data>
+<!--
```
We need to cleanup code. Remove these changes if these files are not needed

sameerz09 committed:
Delete addons/accounts_balances/controllers/controllers.py

This file becomes obsolete once we completely overhaul the data retrieval method from HTTP to API requests.

sameerz09 committed:
Update __init__.py

The models folder is unnecessary, and the primary modification takes place within the controller folder.

sameerz09 committed:
Delete addons/accounts_balances/demo/demo.xml

This file is empty.

sameerz09 committed:
Update __manifest__.py

There is no need for the following files: demo.xml, template.xml, and account_balance_view.xml.

sameerz09 committed:
Update __manifest__.py

There is no need for the following files: demo.xml, template.xml, and account_balance_view.xml.

sameerz09 committed:
Update __init__.py

The file "account_account" is no longer needed.

sameerz09 committed:
Delete addons/accounts_balances/models/account_account.py

The file "account_account" is no longer needed.

sameerz09 committed:
Delete addons/accounts_balances/models/models.py

The file "models.py" is no longer needed.

sameerz09 committed:
Delete addons/accounts_balances/models/__init__.py

The file is no longer needed.

sameerz09 committed:
Delete addons/accounts_balances/views/views.xml

The file  is no longer needed.

sameerz09 committed:
Delete addons/accounts_balances/views/account_balance_view.xml

The file  is no longer needed.

sameerz09 committed:
Update __manifest__.py

The following files have been removed: 'report/template.xml', 'demo/demo.xml' and account_balance_view.xml.

sameerz09 committed:
Delete addons/movments_accounts_balances/demo/demo.xml

no need for demo.xml file

sameerz09 committed:
Delete addons/movments_accounts_balances/views/views.xml

This file is unnecessary.

sameerz09 committed:
Delete addons/movments_accounts_balances/views/account_balance_view.xml

This file is unnecessary.

sameerz09 committed:
Delete addons/movments_accounts_balances/models/models.py

This file is unnecessary.

sameerz09 committed:
Delete addons/movments_accounts_balances/models/account_account.py

This file is unnecessary.

sameerz09 committed:
Delete addons/movments_accounts_balances/models/__init__.py

This file is unnecessary.

sameerz09 committed:
Delete addons/movments_accounts_balances/wizard/__init__.py

This file is unnecessary.

sameerz09 committed:
Delete addons/accounts_balances/wizard/__init__.py

This file is unnecessary.

sameerz09 committed:
Delete addons/accounts_balances/security/ir.model.access.csv

This file is unnecessary.

sameerz09 committed:
Delete addons/accounts_balances/controllers/__init__.py

This file is unnecessary.

sameerz09 committed:
Delete addons/accounts_balances/__manifest__.py

This file is unnecessary.

sameerz09 committed:
Delete addons/accounts_balances/__init__.py

This file is unnecessary.

sameerz09 committed:
Update __init__.py

no need for models folder

sameerz09 reviewed (commented):
All unnecessary folders and files have been removed.

sameerz09 committed:
Update __manifest__.py

no need for account_balance_view.xml

sameerz09 committed:
Update __manifest__.py

sameerz09 committed:
Update ir.model.access.csv

no need for this file

sameerz09 committed:
Delete addons/movments_accounts_balances/security/ir.model.access.csv

no need for this file

sameerz09 committed:
enhance the API to include not only the date but also incorporate a dropdown menu for selecting the account

sameerz09 committed:
Implemented an API to retrieve all accounts from the account.account module

sameerz09 committed:
Select account_id fixing issue

sugampandey commented on addons/movments_accounts_balances/controllers/controllers.py:
```diff
@@ -0,0 +1,37 @@
+    _inherit = 'account.account'
+
+    @api.model
+    def getbalance(self, target_date_start, target_date_end, selected_account_id):
```
Start date can come in blank. If it is blank, then do not put a condition for start date, just use the condition for end date

End date should not be blank.  Handle with an error message if end date comes in blank

sameerz09 committed:
enhance the API to include not only the date but also incorporate a dropdown menu for selecting the account

sugampandey commented on addons/movments_accounts_balances/controllers/controllers.py:
```diff
@@ -0,0 +1,60 @@
+                           l.date AS date, COALESCE(SUM(debit - credit), 0) AS balance
+                    FROM account_move_line AS l
+                    LEFT JOIN account_account AS a ON l.account_id = a.id
+                    WHERE l.account_id = %s AND l.date >= %s AND l.date <= %s
```
Start date can come in blank. If it is blank, then do not put a condition for start date, just use the condition for end date:
```python
WHERE l.account_id = %s AND l.date <= %s
```

sameerz09 committed:
Apply query 2 when start_date is not provided

sameerz09 committed:
Delete addons/movments_accounts_balances/controllers/__init__.py
